content/content_scripts/popup.js

```diff
@@ -0,0 +1,180 @@
+      console.error("Error registering:", error);
+      statusDiv[0].textContent = "An error occurred while registering";
+    }
+  });
+
+  // Event listener for login button
+  loginForm.addEventListener("submit", async (event) => {
+    event.preventDefault();
+    const pass = loginPassword.value;
+
+    statusDiv[1].textContent = "";
+
+    try {
+      // Send logout request to background script
+      const response = await new Promise((resolve, reject) => {
+        chrome.runtime.sendMessage(
+          { action: "login", password: pass },
+          (response) => {
+            if (chrome.runtime.lastError) {
+              reject(chrome.runtime.lastError);
+            } else {
+              resolve(response);
+            }
+          }
+        );
+      });
+      if (response && response.success === true) {
+        statusDiv[1].textContent = "Logged in successfully!";
+        handleServiceWorkerInjection();
+        await updatePopupContent();
+      } else {
+        statusDiv[1].textContent = response.error;
+        passwordInput.value = "";
+      }
+    } catch (error) {
+      // console.error('Error logging in:', error);
+      statusDiv[1].textContent = "An error occurred while logging in.";
+    }
+  });
+
+  // Event listener for logout button
+  logoutForm.addEventListener("submit", async (event) => {
+    event.preventDefault();
+    const pass2 = logoutPassword.value;
+
+    statusDiv[2].textContent = "";
+
+    try {
+      // Send logout request to background script
+      const response = await new Promise((resolve, reject) => {
+        chrome.runtime.sendMessage(
+          { action: "logout", password: pass2 },
+          (response) => {
+            if (chrome.runtime.lastError) {
+              reject(chrome.runtime.lastError);
+            } else {
+              resolve(response);
+            }
+          }
+        );
+      });
+
+      if (response && response.success === true) {
+        statusDiv[2].textContent = "Logged out successfully!";
+        await updatePopupContent();
+      } else {
+        statusDiv[2].textContent = response.error;
+        passwordInput.value = "";
+      }
+    } catch (error) {
+      console.error("Error logging out:", error);
+      statusDiv[2].textContent = "An error occurred while logging out.";
+    }
+  });
+
+  //an array to store toggle elements and their checked status for service worker injection
+  const toggles = [
+    { element: socialToggle, checked: false },
+    { element: gamingToggle, checked: false },
+  ];
+
+  // Function to handle service worker injection based on toggle status and update the toggles array
+  const handleServiceWorkerInjection = async () => {
+    const checkedToggles = toggles.filter((toggle) => toggle.checked);
+    await injectServiceWorker(toggles,checkedToggles);
+  };
+
+  // Add event listeners to toggle elements
+  toggles.forEach((toggle) => {
+    toggle.element.addEventListener("change", function () {
```

Comment:
we do not need an event listener if we are using the state of check box on submit